File: c++/pnc1d/eeq.cpp
#include <iostream>
#include <cmath> // Para funções matemáticas

const double h = 6.62607015e-34; // Constante de Planck em Joules.s

double calcularEnergia(double L, int n, double m) {
    return (std::pow(n, 2) * std::pow(h, 2)) / (8 * m * std::pow(L, 2));
}

int main() {
    double L; // Comprimento da caixa em metros
    int n;    // Número quântico principal
    double m; // Massa da partícula em kg

    // Entrada dos parâmetros
    std::cout << "Use xe-y para notações cientificas se necessário." << std::endl;
    std::cout << "Digite o comprimento da caixa (L) em metros: ";
    std::cin >> L;

    std::cout << "Digite o número quântico principal (n): ";
    std::cin >> n;

    std::cout << "Digite a massa da partícula (m) em kg: ";
    std::cin >> m;

    // Verificar se os valores são válidos
    if (L <= 0 || n <= 0 || m <= 0) {
        std::cerr << "Erro: L, n e m devem ser valores positivos!" << std::endl;
        return 1;
    }

    // Cálculo da energia
    double energia = calcularEnergia(L, n, m);

    // Exibição do resultado
    std::cout << "A energia do estado quantico (En) é: " << energia << " Joules." << std::endl;

    return 0;
}

File: html_pt/js/eeq.js
const h = 6.62607015e-34; // Constante de Planck (J.s)

        function calcularEnergia() {
            // Obter valores dos inputs
            const L = parseFloat(document.getElementById('L').value);
            const n = parseInt(document.getElementById('n').value);
            const m = parseFloat(document.getElementById('m').value);

            // Referências para exibição de resultado e erro
            const resultDiv = document.getElementById('result');
            const errorDiv = document.getElementById('error');
            const energiaSpan = document.getElementById('energia');

            // Limpar mensagens anteriores
            resultDiv.style.display = "none";
            errorDiv.style.display = "none";

            // Verificar se os valores são válidos
            if (isNaN(L) || isNaN(n) || isNaN(m) || L <= 0 || n <= 0 || m <= 0) {
                errorDiv.textContent = "Erro: Todos os valores devem ser positivos e válidos!";
                errorDiv.style.display = "block";
                return;
            }

            // Calcular a energia
            const energia = (Math.pow(n, 2) * Math.pow(h, 2)) / (8 * m * Math.pow(L, 2));

            // Exibir o resultado
            energiaSpan.textContent = energia.toExponential(6); // Formatar resultado em notação científica
            resultDiv.style.display = "block";}